crates/uv/tests/it/pip_compile.rs

```diff
@@ -14073,3 +14055,83 @@ fn dynamic_version_source_dist() -> Result<()> {
 
     Ok(())
 }
+
+#[test]
+fn max_python_requirement() -> Result<()> {
+    static EXCLUDE_NEWER: &str = "2024-12-18T00:00:00Z";
+
+    let context = TestContext::new("3.8");
+    let requirements_in = context.temp_dir.child("requirements.in");
+    requirements_in.write_str(indoc::indoc! {r"
+        nox >=2024.04.15
+        nox[uv] >=2024.03.02; python_version >= '3.8'
+    "})?;
+
+    uv_snapshot!(context.filters(), windows_filters=false, context.pip_compile()
+            .env(EnvVars::UV_EXCLUDE_NEWER, EXCLUDE_NEWER)
+            .arg("requirements.in")
+            .arg("--universal")
+            .arg("--python-version")
+            .arg("3.7"), @r###"
+    success: true
+    exit_code: 0
+    ----- stdout -----
+    # This file was autogenerated by uv via the following command:
+    #    uv pip compile --cache-dir [CACHE_DIR] requirements.in --universal --python-version 3.7
+    argcomplete==3.1.2 ; python_full_version < '3.8'
+        # via nox
+    argcomplete==3.5.2 ; python_full_version >= '3.8'
+        # via nox
+    colorama==0.4.6 ; sys_platform == 'win32'
+        # via colorlog
+    colorlog==6.9.0
+        # via nox
+    distlib==0.3.9
+        # via virtualenv
+    filelock==3.12.2 ; python_full_version < '3.8'
+        # via virtualenv
+    filelock==3.16.1 ; python_full_version >= '3.8'
+        # via virtualenv
+    importlib-metadata==6.7.0 ; python_full_version < '3.8'
+        # via
+        #   argcomplete
+        #   nox
+        #   virtualenv
+    nox==2024.4.15 ; python_full_version < '3.8'
+        # via -r requirements.in
+    nox==2024.10.9 ; python_full_version >= '3.8'
+        # via -r requirements.in
+    packaging==24.0 ; python_full_version < '3.8'
+        # via nox
+    packaging==24.2 ; python_full_version >= '3.8'
+        # via nox
+    platformdirs==4.0.0 ; python_full_version < '3.8'
+        # via virtualenv
+    platformdirs==4.3.6 ; python_full_version >= '3.8'
+        # via virtualenv
+    tomli==2.0.1 ; python_full_version < '3.8'
+        # via nox
+    tomli==2.2.1 ; python_full_version >= '3.8' and python_full_version < '3.11'
+        # via nox
+    typing-extensions==4.7.1 ; python_full_version < '3.8'
+        # via
+        #   importlib-metadata
+        #   nox
+        #   platformdirs
+    uv==0.5.10 ; python_full_version >= '3.8'
+        # via nox
+    virtualenv==20.26.6 ; python_full_version < '3.8'
+        # via nox
+    virtualenv==20.28.0 ; python_full_version >= '3.8'
+        # via nox
+    zipp==3.15.0 ; python_full_version < '3.8'
+        # via importlib-metadata
```

Comment:
Lots of splits, but this _is_ correct... A lot of packages dropped support for Python 3.7, but the user still wants to resolve for it.